packages/pwa-kit-react-sdk/src/ssr/universal/components/switch/index.jsx

```diff
@@ -53,4 +45,27 @@ Switch.propTypes = {
     preloadedProps: PropTypes.object
 }
 
+const RoutesConsumer = ({appState}) => {
+    const {routes} = useRoutes()
+
+    return (
+        <RouterSwitch>
+            {routes.map((route, i) => {
+                const {component: Component, ...routeProps} = route
+                return (
+                    <Route key={i} {...routeProps}>
+                        <UIDFork>
+                            <Component preloadedProps={appState.pageProps} />
+                        </UIDFork>
+                    </Route>
+                )
+            })}
+        </RouterSwitch>
+    )
+}
+
+RoutesConsumer.propTypes = {
+    appState: PropTypes.object.isRequired
+}
+
```

Comment:
in order to move the Routes provider/context to a separate file I needed to pull out the Route logic in Switch to it's own component so we can use the provider correctly and reference the routes in the context

Reply:
I'm not sure I fully understand this comment. Does this RoutesConsumer get used anywhere else in the project?